service/smtp-service.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+var fs = require('fs');
+var templateEngine = require('handlebars');
+var nodemailer = require('nodemailer');
+var smtpTransport = require('nodemailer-smtp-transport');
+var promise = require('bluebird');
+
+promise.promisifyAll(fs);
+
+var SmtpService = function SmtpService(emailModel) {
+  this.dataModel = emailModel || {};
+  this.template = null;
+  this.compiledTemplate = null;
+  this.populatedTemplate = '';
+
+  this.createTransport();
+};
+
+/**
+ * Create our transport mechanism from config and return it
+ */
+SmtpService.prototype.createTransport = function createTransport() {
+  var config = require('../config/smtp.json');
+
+  this.emailTransport = nodemailer.createTransport(
+    smtpTransport(
+      {
+        host: config.host,
+        port: config.port,
+        auth: {
+          user: config.auth.user,
+          pass: config.auth.password
+        }
+      }
+    )
+  );
+};
+
+/**
+ * Load an email template specified in the body of the model
+ */
+SmtpService.prototype.loadTemplate = function loadTemplate() {
+  return fs.readFileAsync(this.dataModel.template).then(function bindTemplate(data) {
+    this.template = data;
+    return this.template;
+  }.bind(this))
+  .catch(function exceptionHandler(error) {
+    throw error;
+  });
+};
+
+/**
+ * Work out magic and return the processed template
+ * @returns {*|String}
+ */
+SmtpService.prototype.prepareContent = function prepareContent() {
+
+  return fs.readFileAsync(this.dataModel.template).then(function bindData(data) {
+    this.template = data;
+    this.compiledTemplate = templateEngine.compile(this.template.toString());
```

Comment:
`compile` expects a string `fs` returns a bytestream, changed the casting so it does it prior to pushing it into the compile `method`